### ISSUE TITLE
Change Request: context.getOriginalFilename() method

### ISSUE THREAD

**Issue by mmkal** [eslint/eslint#17768]
### ESLint version

v8.53.0

### What problem do you want to solve?

ESLint currently offers a `context.getFilename()` method for plugin authors. But, the value returned by it can be affected by processors (e.g. eslint-plugin-markdown). This means if another plugin does something like this:

```ts
export const rules = {
  myrule: {
    create: context => {
      const text = require('fs').readFileSync(context.getFilename(), 'utf8')
      if (text.includes('f')) {
        context.report({ message: `Don't use the letter f`, ... })
      }
    }
  }
}
```

It will usually work, but it will fail unexpectedly if the user uses a processor which transforms the filename from `/foo/bar.md` into `/foo/bar.md/0_0.js`.

### What do you think is the correct solution?

A new method `context.getOriginalFilename()` which always the returns the "real" filepath. If no processor is used, it's the same as `context.getFilename()`.

### Participation

- [X] I am willing to submit a pull request for this change.

### Additional comments

Hopefully less of a big request than https://github.com/eslint/eslint/issues/17724 which highlights a real problem, but would require a big refactor, it sounds like.

See https://github.com/mmkal/eslint-plugin-codegen/issues/7 for a bug report in an existing plugin, where `eslint-plugin-codegen` is currently relying on a processor. I'm planning on migrating away from a processor and to using `fs.readFileSync` since #17724 was closed.

Note: in the case of `eslint-plugin-markdown` and `eslint-plugin-codegen` it's probably going to be possible to hack around by just doing `context.getFilename().replace(/\/\d+_\d+\.js$/, '')`. But that is pretty brittle and not generalisable.

**Comment by mdjermanovic** [comment]
Hi @mmkal, thanks for the issue!

> A new method `context.getOriginalFilename()` which always the returns the "real" filepath. If no processor is used, it's the same as `context.getFilename()`.

This feature already exists as `context.physicalFilename`/`context.getPhysicalFilename()` (just note that the latter, function form is deprecated and will be removed in the future).

https://eslint.org/docs/latest/extend/custom-rules#the-context-object

**Comment by mmkal** [comment]
Great, thanks @mdjermanovic!
